### DISCUSSION TITLE
xDrip could enable Bluetooth even in airplane mode

### DISCUSSION THREAD

**Discussion by Navid200** [NightscoutFoundation/xDrip#3276] [Show and tell]
If airplane mode was enabled, xDrip would not enable Bluetooth. You would have to enable it yourself.  

As of the last Nightly release (January 5, 2024), xDrip will enable Bluetooth if it needs to communicate with something through it (Collecting from a CGM data source, a Bluetooth meter, ...), even if airplane mode is enabled.

**Comment by gitanO0** [comment]
Thank you!

**Comment by Navid200** [comment]
xDrip needs Bluetooth to do its job if it is set to collect from a CGM device or from a Bluetooth glucose meter or drive a watch through Bluetooth.
xDrip enables Bluetooth, even before this change, if it needs it.  Only in airplane mode, it did not.  Now, it does even in airplane mode.    

Even before this change, you could manually enable Bluetooth after entering airplane mode.
You still can.
But, before this change, if something caused Bluetooth to be disabled in airplane mode, xDrip would not enable Bluetooth.  Now, it does if it needs it.

When you enable airplane mode, Bluetooth is disabled automatically by Android until Android 14.
Android 14 also does the same!  However, if you then enable Bluetooth in airplane mode, Android 14 will remember that.  The next time you enable airplane mode, it will not disable Bluetooth.

**What if I want to keep Bluetooth disabled?**  
Then, you can disable any communication in xDrip that relies on Bluetooth.  If you use xDrip only to follow someone using xDrip sync, xDrip will not need Bluetooth and will not enable it.  
Alternatively, you can uninstall xDrip.
